Studio-Lovelies/GG-JointJustice-Unity
# Show dialogue choice menus again

## Summary
Resolves a regression discovered by @PhoebeMitchell: Choice menus outside investigations were no longer showing up and players automatically continued as if they chose the first action.

## Before/after screenshots and/or animated gif

Running…

```ink
&SCENE:TMPH_Court
&SET_ACTOR_POSITION:Defense,Arin
&JUMP_TO_POSITION:Defense
&SET_POSE:Thinking
&THINK:Arin
Hm...
&HIDE_TEXTBOX

+ [Object!]
    -> playerObjects
+ [Nah, we gucci]
    -> playerDoesntObject

=== playerObjects ===
&OBJECTION:Arin
&PLAY_SFX:deskSlam
&PLAY_EMOTION:DeskSlamAnimation
&SPEAK:Arin
My client's behavior during the livestream isn't suspicious in the least!

-> END

=== playerDoesntObject ===
&SPEAK:Arin
Nah, it's fine. Polishing some coins doesn't prove he's guilty.
-> END
```
### Before

https://github.com/user-attachments/assets/48a6ad77-e5c7-409b-8912-3e8474ec8251

### After

https://github.com/user-attachments/assets/42ba2277-c641-46ee-8085-bd471733d0e8

## Testing instructions
1. Open the `Game` scene
2. Set the `GameStarter` to `ChoiceMenu`
3. Notice the choice menu opening after advancing the game
4. Copy the `ChoiceMenu` script to your Desktop
5. Switch to `develop`
6. Copy the `ChoiceMenu` script back to Unity and select it again
7. Notice no choice menu opening and Arin immediately objecting after advancing the game

## Additional information
The issue stems from [this change](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/commit/720a14851fa2a739fe079a964f050512fc9667d7) to `ChoiceMenu.cs` introduced by me as part of #460.

The InvestigationState injects logic into the ChoiceMenu to handle things like the `Move` preview image or checkboxes for `Talk` options. This requires the menu to be populated right after calling `ChoiceMenu.Initialise()`.

However this makes the `ChoiceMenu` immediately listen to input. "The player pressing `X` to advance to the choices", becomes "the player pressing `X` to confirm a choice" as well, as they both happen in the same frame.

To resolve this, I have passed the logic from the investigation state into a callback the ChoiceMenu calls after the buttons are created after a one frame delay. This resolves the regression while retaining the new functionality. (79942516217eab8e93ad7ad87831ec0880fdef98)

I've also added an automated test to catch this in the future. Running this test on `develop` right now will reliably fail, while it succeeds on this branch. (16793b45c9161a2f10b8031cf3cd9f1d06e835b7)

A different test failed after my initial submission of this PR, as I didn't migrate all parts of code to the callback; this was remedied in 0ca1a92d53b045df7bc28ca5839656d0b6e1d50d.


- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:
